src/ORM/DataObject.php

```diff
@@ -847,7 +847,7 @@ public function defineMethods()
      */
     public function exists()
     {
-        return (isset($this->record['ID']) && $this->record['ID'] > 0);
```

Comment:
Just had a look around in DataObject __construct and hydrate() - looks like there's some options to create a record, with an ID, but not have it in the database.  You could possibly argue that exists() and isInDB() *should* have different logic, though possibly hydrating with an ID for a record that isn't in the database is too much of a niche case to consider.  Do you have any views on this?

Reply:
The current implementation of `isInDb()` will ultimately give the same Boolean result as this method in that scenario, assuming the hydrated data includes a numeric ID:
`return is_numeric($this->ID) && $this->ID > 0;`

Regardless of that, I think hydrating a record that is not in the DB without immediately writing that record is an unlikely scenario.

Reply:
Yes, also just having `<DataObject>->exists()` return true for a non-persisted DataObject is just, weird

While this is an edge case, it is a functional change of sorts. Do you think this is worth pointing to `5` instead or is this just being overly cautious?

Reply:
Personally I think that's overly cautious - the implementations of these methods are so similar it reads to me as though they're intended to do the exact same thing. Any difference in functionality imo is unintentional.